deploy-service/common/src/main/java/com/pinterest/deployservice/handler/DeployHandler.java

```diff
@@ -291,7 +291,7 @@ String internalDeploy(EnvironBean envBean, DeployBean deployBean) throws Excepti
         EnvironBean updateEnvBean = new EnvironBean();
         updateEnvBean.setDeploy_id(deployId);
         updateEnvBean.setDeploy_type(deployBean.getDeploy_type());
-        updateEnvBean.setStage_type(envBean.getStage_type());
+        updateEnvBean.updateStage_type(envBean.getStage_type());
```

Comment:
Looks good, but this changes fixes already present bug in the project. Please add unit tests that proofs that fix really works and we will have automated testing which will check that we will not have the same problem in the future.

Reply:
Thanks, just added some. I ended up moving the business logic outside of the bean. I did add some tests from the API side, but didn't do the ones from db -> object as they wouldn't be relevant anymore.